packages/yarnpkg-core/sources/Cache.ts

```diff
@@ -120,15 +120,25 @@ export class Cache {
   }
 
   async setup() {
+    // mkdir may cause write operations even when directories exist. To ensure that the cache can be successfully used
+    // on read-only filesystems, only run mkdir when not running in immutable mode.
```

Comment:
Is this comment correct? How can `mkdir` make write operations when the directories exist? Isn't the problem rather than you didn't have the folder, and thus the write operation failed? 🤔

If the comment is correct, perhaps it'd be worth posting it on the Node repo as well along with a repro, that sounds really strange.

Reply:
Digging a bit it may be related to the behaviour described here:
https://bug-coreutils.gnu.narkive.com/jKTgWRAO/mkdir-p-and-erofs

Reply:
I'm going to see if I can put together a minimal repro; certainly `mkdirp()` works where `mkdir(, {recursive: true})` doesn't.

Reply:
Ah...the critical bit of information I forgot to include is that this is on Windows...

Reply:
https://github.com/mrginglymus/mkdir-doom <- minimal repro. So it's definitely a node bug, but I would appreciate if this (or similar) workaround could be included in yarn to save me having to patch the offending line.

Reply:
https://github.com/nodejs/node/issues/38347